cache.go

```diff
@@ -208,7 +219,27 @@ func (r *Cache) Resolve(q *dns.Msg, ci ClientInfo) (*dns.Msg, error) {
 
 	// Put the upstream response into the cache and return it. Need to store
 	// a copy since other elements might modify the response, like the replacer.
-	r.storeInCache(q, a.Copy())
+	// For Redis backends with async-set-on-miss enabled, store asynchronously
+	// to reduce miss-path latency.
+	if rb, ok := r.backend.(*redisBackend); ok && rb.opt.AsyncSetOnMiss {
```

Comment:
Not a fan of doing the type cast here. Ideally, this code should go into cache-redis.go (in `Store()`). Then there's no casting necessary and the cache doesn't need to know/care about what kind of backend is being used.